fuel-core/src/executor.rs

```diff
@@ -102,34 +100,42 @@ impl Executor {
 
         // setup and execute block
         let current_height = db.get_block_height()?.unwrap_or_default();
-        let current_hash = db.get_block_id(current_height)?.unwrap_or_default();
         let new_block_height = current_height + 1u32.into();
 
-        let mut block = FuelBlock {
-            header: FuelBlockHeader {
-                height: new_block_height,
-                parent_hash: current_hash,
-                time: Utc::now(),
-                ..Default::default()
+        let block = PartialFuelBlock::new(
+            PartialFuelBlockHeader {
+                application: FuelApplicationHeader {
+                    // TODO: This should not be default
+                    da_height: Default::default(),
+                    generated: Default::default(),
+                },
+                consensus: FuelConsensusHeader {
+                    // TODO: This should not be default
+                    prev_root: Default::default(),
+                    height: new_block_height,
+                    time: Utc::now(),
+                    generated: Default::default(),
+                },
+                metadata: Default::default(),
             },
-            transactions: txs.into_iter().map(|t| t.as_ref().clone()).collect(),
-        };
+            txs.into_iter().map(|t| t.as_ref().clone()).collect(),
+        );
         // immediately execute block
-        self.execute(&mut block, ExecutionMode::Production).await?;
+        self.execute(block, ExecutionMode::Production).await?;
         Ok(())
     }
 
     #[tracing::instrument(skip(self))]
     pub async fn execute(
         &self,
-        block: &mut FuelBlock,
+        mut block: PartialFuelBlock,
         mode: ExecutionMode,
-    ) -> Result<(), Error> {
-        // Compute the block id before execution, if mode is set to production just use zeroed id.
-        let pre_exec_block_id = match mode {
-            ExecutionMode::Production => Default::default(),
-            ExecutionMode::Validation => block.id(),
-        };
+    ) -> Result<FuelBlock, Error> {
+        // // Compute the block id before execution, if mode is set to production just use zeroed id.
+        // let pre_exec_block_id = match mode {
+        //     ExecutionMode::Production => Default::default(),
+        //     ExecutionMode::Validation => block.id(),
+        // };
```

Comment:
This doesn't make sense anymore because a partial block doesn't have an id.

Reply:
Hmm actually I think it might've been correct to use the mutated tx bytes from the VM. This is because this is our only way to obtain a commitment to the resultant state of executing the tx, whereas the transaction ID doesn't include any of the malleable fields. Without a commitment to the actual transaction bytes, things like change amounts calculated by the VM wouldn't be provable between nodes or on Ethereum.

Reply:
Ok added this test back.